src/hotspot/cpu/aarch64/c1_LIRAssembler_aarch64.cpp

```diff
@@ -2984,7 +2984,25 @@ void LIR_Assembler::membar_loadstore() { __ membar(MacroAssembler::LoadStore); }
 void LIR_Assembler::membar_storeload() { __ membar(MacroAssembler::StoreLoad); }
 
 void LIR_Assembler::on_spin_wait() {
-  Unimplemented();
+  switch (VM_Version::pause_impl_desc().inst()) {
+    case NOP:
+      for (unsigned int i = 1; i < VM_Version::pause_impl_desc().inst_count(); ++i) {
```

Comment:
Good catch. It is a copy-paste error.
Is there any method to test C1 generated assembly code?

Reply:
You could do it the same way as hotspot/jtreg/compiler/c2/aarch64/TestVolatiles.java, i.e. spawn a subtask and parse the output dump. It's very fiddly, though.

Reply:
Yes, I used it as an example when I was writing tests for the PR. It works only for C2 because it relies on C2 `XX:+PrintOptoAssembly`. I haven't found anything similar for C1.

Reply:
Fixed.

Reply:
`-XX:+PrintAssembly`

Reply:
To have assembly instructions in `-XX:+PrintAssembly` output `hsdis` needs to be provided:
```
  0x0000ffff61ba2b5c: ;   {metadata({method} {0x0000000800466ab8} 'isLatin1' '()Z' in 'java/lang/String')}
  0x0000ffff61ba2b5c: 0857 8dd2 | c808 a0f2 | 0801 c0f2 | e807 00f9
 ;; 0xFFFFFFFFFFFFFFFF
  0x0000ffff61ba2b6c: 0800 8092 | e803 00f9

  0x0000ffff61ba2b74: ;   {runtime_call counter_overflow Runtime1 stub}
```

 However it can help to skip to the place where instructions are expected and to check instructions' hex code.

Reply:
I rewrote a test to parse `XX:+PrintAssembly` hex instructions.